src/plugins/security_oss/public/plugin.ts

```diff
@@ -17,27 +17,51 @@
  * under the License.
  */
 
-import { CoreSetup, CoreStart, Plugin, PluginInitializerContext } from 'src/core/public';
+import {
+  Capabilities,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  PluginInitializerContext,
+} from 'src/core/public';
 import { ConfigType } from './config';
 import {
   InsecureClusterService,
   InsecureClusterServiceSetup,
   InsecureClusterServiceStart,
 } from './insecure_cluster_service';
+import { AppStateService } from './app_state';
+
+export interface SavedObjectTypeAnonymousAccess {
+  /**
+   * Indicates whether anonymous user can access particular Saved Object type (e.g. dashboard, map etc.).
+   */
+  canAccess: boolean;
+  /**
+   * A map of query string parameters that should be specified in URL so that anonymous user can use
+   * to automatically log in to Kibana and access particular Saved Object type.
+   */
+  accessURLParameters: Record<string, string> | null;
+}
 
 export interface SecurityOssPluginSetup {
   insecureCluster: InsecureClusterServiceSetup;
 }
 
 export interface SecurityOssPluginStart {
   insecureCluster: InsecureClusterServiceStart;
+  anonymousAccess: {
```

Comment:
__note:__ I'll discuss the final API shape with AppArch, but I don't expect it to change too much. It feels the best option would be to give Share plugin consumers entire capabilities object so that they can the check on their own.

Reply:
I agree, I don't see a need to filter out the capabilities object on our side.